lib/ansible/module_utils/powershell.ps1

```diff
@@ -225,3 +225,20 @@ Function Get-FileChecksum($path)
     }
     return $hash
 }
+
+Function Get-PendingRebootStatus
+{
+    # Check if reboot is required, if so notify CA. The MSFT_ServerManagerTasks provider is missing on client SKUs
+    #Function returns true if computer has a pending reboot
+    $featureData = invoke-wmimethod -EA Ignore -Name GetServerFeature -namespace root\microsoft\windows\servermanager -Class MSFT_ServerManagerTasks
+    $regData = Get-ItemProperty "HKLM:\SYSTEM\CurrentControlSet\Control\Session Manager" "PendingFileRenameOperations" -EA Ignore
+    if(($featureData -and $featureData.RequiresReboot) -or $regData)
+    {
+        return $True
+    }
+    else 
+    {
+        return $False
+    }
+
+}
```

Comment:
@trondhindenes After a couple rounds of "WTH?!" with this on my Win10 box (it took 3 reboots to get to "false"- Chrome kept putting things in the PendingFileRenameOperations key), I'm thinking we might want an optional ref arg where we'll return a list of the reasons. I also have a couple more locations to look (win updates, CBS, etc), so with all those places, it'd probably be good to know why a reboot is pending.